src/auto-evo/steps/ModifyExistingSpecies.cs

```diff
@@ -0,0 +1,327 @@
+
+        foreach (var variant in viableVariants.Select(x => x.Item1))
+        {
+            MutationLogicFunctions.NameNewMicrobeSpecies(variant, baseSpecies);
+
+            var oldColour = variant.Colour;
+
+            var redShift = (random.NextDouble() - 0.5f) * Constants.AUTO_EVO_COLOR_CHANGE_MAX_STEP;
+            var greenShift = (random.NextDouble() - 0.5f) * Constants.AUTO_EVO_COLOR_CHANGE_MAX_STEP;
+            var blueShift = (random.NextDouble() - 0.5f) * Constants.AUTO_EVO_COLOR_CHANGE_MAX_STEP;
+
+            variant.Colour = new Color(Mathf.Clamp((float)(oldColour.R + redShift), 0, 1),
+                Mathf.Clamp((float)(oldColour.G + greenShift), 0, 1),
+                Mathf.Clamp((float)(oldColour.B + blueShift), 0, 1));
+        }
+
+        return GetTopMutations(baseSpecies, viableVariants, amount, cache, selectionPressures).Select(x => x.Item1)
+            .ToList();
+    }
+
+    public static List<Tuple<MicrobeSpecies, float>> PruneMutations(MicrobeSpecies baseSpecies,
+        List<Tuple<MicrobeSpecies, float>> mutated, SimulationCache cache, List<SelectionPressure> selectionPressures)
+    {
+        var outputSpecies = new List<Tuple<MicrobeSpecies, float>>();
+
+        foreach (var potentialVariant in mutated)
+        {
+            var combinedScores = 0.0;
+            foreach (var pastPressure in selectionPressures)
+            {
+                var newScore = pastPressure.CacheScore(potentialVariant.Item1, cache);
+                var oldScore = pastPressure.CacheScore(baseSpecies, cache);
+
+                // Break if mutation fails a new pressure
+                if (newScore <= 0 && oldScore > 0)
+                {
+                    combinedScores = -1;
+                    break;
+                }
+
+                combinedScores += pastPressure.WeightedComparedScores(newScore, oldScore);
+            }
+
+            if (combinedScores > 0)
+            {
+                outputSpecies.Add(potentialVariant);
+            }
+        }
+
+        return outputSpecies;
+    }
+
+    public static IEnumerable<Tuple<MicrobeSpecies, float>> GetTopMutations(MicrobeSpecies baseSpecies,
+        List<Tuple<MicrobeSpecies, float>> mutated, int amount, SimulationCache cache,
+        List<SelectionPressure> selectionPressures)
+    {
+        return mutated.OrderByDescending(x =>
+                selectionPressures.Select(pressure =>
+                    pressure.CacheScore(x.Item1, cache) / pressure.CacheScore(baseSpecies, cache) *
+                    pressure.Strength).Sum())
+            .ThenBy(x => x.Item2).Take(amount);
+    }
+
+    public bool RunStep(RunResults results)
+    {
+        // TODO: Have this be passed in
+        var random = new Random();
+
+        var oldMiche = results.MicheByPatch[Patch];
+        var oldOccupants = oldMiche.GetOccupants().Distinct().ToList();
+
+        // TODO: Possibly make this a performance setting?
+        const int possibleMutationsToTryPerSpecies = 3;
```

Comment:
Isn't there an auto-evo config variable for this already? That should be hooked up to here.